graftm/pplacer.py

```diff
@@ -50,16 +50,23 @@ def jplace_split(self, jplace_file, alias_hash):
 
         # Load the placement file
         placement_file = json.load(open(jplace_file))
-
+        
         # Parse the placements based on unique identifies appended to the end
         # of each read
         for placement in placement_file['placements']: # for each placement
             hash = {} # create an empty hash
             for alias in alias_hash: # For each alias, append to the 'place' list each read that identifier
-                hash = {'p': placement['p'],
-                        'nm': [nm for nm in placement['nm'] if nm[0].split('_')[-1] == alias]}
-                alias_hash[alias]['place'].append(hash)
-
+                nm_list = []
+                p = placement['p']
+                for nm in placement['nm']:
+                    if nm[0].split('_')[-1] == alias:
```

Comment:
If > 1 file is given to GraftM it "barcodes" each sequence so that the file of origin (i.e. the alias) for each sequence is always known. This way we only need to run pplacer once, as opposed to running it for every input file. I'll add the comment in.